WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelCreatePackageViewModel.kt

```diff
@@ -45,21 +56,24 @@ class ShippingLabelCreatePackageViewModel @Inject constructor(
         )
     }
 
-    /**
-     * Saving more data than necessary into the SavedState has associated risks which were not known at the time this
-     * field was implemented - after we ensure we don't save unnecessary data, we can replace @Suppress("OPT_IN_USAGE")
-     * with @OptIn(LiveDelegateSavedStateAPI::class).
-     */
-    @Suppress("OPT_IN_USAGE")
-    val viewStateData = LiveDataDelegate(savedState, ShippingLabelCreatePackageViewState())
+    fun onSelectedPageChanged(selectedTab: PackageType) {
+        selectedTabType.update { selectedTab }
+    }
 
-    @Parcelize
-    data class ShippingLabelCreatePackageViewState(
-        val selectedTab: PackageType = PackageType.CUSTOM
-    ) : Parcelable
+    fun onDoneButtonClicked() {
+        launch { _creationDoneFlow.emit(OnDoneButtonClicked(selectedTab = selectedTabType.value)) }
+    }
 
     enum class PackageType {
         CUSTOM,
-        SERVICE
+        SERVICE;
+
+        companion object {
+            fun fromOrdinal(ordinal: Int) = entries.first { it.ordinal == ordinal }
```

Comment:
np: Accessing the value using the ordinal (position) should be faster than iterating on the entries list. WDYT?

```suggestion
            fun fromOrdinal(ordinal: Int) = entries[ordinal]
```

Reply:
I don't even know why I submitted this terrible way of ordinal access 😅 Thanks, @atorresveiga